homeassistant/components/modbus/__init__.py

```diff
@@ -282,6 +282,7 @@
                 vol.All(
                     {
                         CONF_ADDRESS: cv.positive_int,
+                        vol.Optional(CONF_WRITE_REGISTERS, default=False): cv.boolean,
```

Comment:
A much better option is to have CONF_ADDRESS be either an int or an array, no need for an extra parameter.

Please remember this needs to be copied for FAN_MODE.

Reply:
Hi @janiversen thansk for the review. With regards to the suggestion, I followed the same approach already implemented for the other parameters, in example [HVAC Mode](https://www.home-assistant.io/integrations/modbus/#write_registers), target temp and [hvac on off](https://www.home-assistant.io/integrations/modbus/#write_registers) In those cases, the address is an integer and using that option the user can force the system to use te modbus function write multiple registers (0x10) instead of the single (0x06). So I considered this approach as the "suggested one". Why  we should change it only for fan mode?

Reply:
Because it simplifies the code, and that is the direction we are moving towards....of course not all of the old code is changed at this point in time.

Reply:
And if you look at e.g. the service set point, it does exactly that, can be called with an int or an array.

Reply:
"Coherent" is not an objective in itself, especially not if it implies expanding the code.

Please do not forget, your change is currently not working if you want to use modes....you need to do exact the same for each mode...and that explodes the code.

Reply:
> * if list contains the exact number of variants (in this case modes) use the correct int as address
> 
> * if the entry in list is a list with 1 entry use write_registers.

I really appreciate the condivision of your vision and I imagine that this proposal will make really complex the code. I know that my opinion is not important, but I miss something. Why do you want to move the code there? 
I mean, if we consider the Fan Mode example, the only need is to have the CONF_ADDRESS managed as int or list of one single int, because do you imagine that also the values will be expressed as list? What is the real use case that I miss?

Reply:
Ok. So, should I replace actual implementation by managing int/list for CONF_ADDRESS?

Reply:
I was just applying the change when I thought that before to do it, probably I need to more information of the future because here we have mixed CONF_ADDRESS and modes.. 
Actually the component uses only an int (CONF_ADDRESS) for the register, but we could have a list for the values. This because the pb_call accepts a list only for the value. 
Generally speaking, the real use cases could include also the the need to write real multiple registers ans so you could have the need to manage a list also as registers. This could increase the complexity of the code, of course, because the options to manage increases. 
@janiversen, considering the actual code, probably to enable the write_registers support I should keep the CONF_ADDRESS as an int and manage int or list for each of the modes. 
If you agree, I change the code in this way, instead of CONF_ADDRESS else please advise me

Reply:
the better option is to have CONF_ADDRESS as an int (write_register) or as a list of one (write_registers). That is the upcoming change for all write_registers